### PR TITLE
Null safety in ErrorMessageDto.

### PR DESCRIPTION
**Pull request by JanDoerrenhaus**

### CODE REVIEW

**Review by schroenser — APPROVED** [review]
Note that Sonar complains about the third `if`, where the second half `otherGlobalParameters != null` is always true, due to the first `if`. In my opinion, this would be harder to read, if we performed that change.

File: core/src/main/java/com/github/mizool/core/rest/errorhandling/ErrorMessageDto.java [hunk 1]
```diff
@@ -30,21 +30,45 @@ public ErrorMessageDto combineWith(ErrorMessageDto other)
...
+            result = otherGlobalParameters;
+        }
+        else
+        {
+            Set<String> combinedKeys = Sets.union(globalParameters.keySet(), otherGlobalParameters.keySet());
 
-        Set<String> combinedKeys = Sets.union(globalParameters.keySet(), otherGlobalParameters.keySet());
+            result = combinedKeys.stream()
+                .collect(ImmutableMap.toImmutableMap(key -> key,
+                    key -> resolveValueInOrder(key, otherGlobalParameters, globalParameters)));
```
Comment (bannmann) [inline core/src/main/java/com/github/mizool/core/rest/errorhandling/ErrorMessageDto.java:68]: Please rename lambda arg `key` to `value`.

(Yes, this was already wrong before.)

**Comment by bannmann** [comment]
> Note that Sonar complains about the third `if`, where the second half `otherGlobalParameters != null` is always true, due to the first `if`. In my opinion, this would be harder to read, if we performed that change.

I agree, let's not follow Sonar's advice here.

We *could* probably get rid of all the `if`s by always going through the `combinedKeys` approach. We would only need to do `if (parameters == null) parameters = Map.of();` for both maps at the start of the method. But I'm fine with leaving it like it is.

**Review by bannmann — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/github/mizool/core/rest/errorhandling/ErrorMessageDto.java (hunk 1 above)
Comment (JanDoerrenhaus) [reply core/src/main/java/com/github/mizool/core/rest/errorhandling/ErrorMessageDto.java:68]: I don't get it. We are streaming over combined keys, so does the lambda arg name `key` not make perfect sense here?

Comment (schroenser) [reply core/src/main/java/com/github/mizool/core/rest/errorhandling/ErrorMessageDto.java:68]: Yes it does. :D Both the key function and the value function of the map collector get the `key` as parameter.

Comment (bannmann) [reply core/src/main/java/com/github/mizool/core/rest/errorhandling/ErrorMessageDto.java:68]: Agreed, I was confused.

**Review by bannmann — APPROVED** [review]

**Pull request merged**